pipeline-runner/R/handle_data.R

```diff
@@ -51,17 +42,64 @@ reload_scdata_from_s3 <- function(pipeline_config, experiment_id, task_name, tas
   obj <- readRDS(rawConnection(body))
   return(obj)
 }
+reload_scdata_list_from_s3 <- function (s3, experiment_id) {
+  bucket <- pipeline_config$source_bucket
+  objects <- s3$list_objects(
+    Bucket = bucket,
+    Prefix = experiment_id
+  )
+  samples <- objects$Contents
+
+  scdata_list <- list()
+  for (sample in samples) {
+    key <- sample$Key
+
+    c(body, ...rest) %<-% s3$get_object(
+      Bucket = bucket,
+      Key = paste(key, sep = "/")
+    )
+    obj <- readRDS(rawConnection(body))
+    sample_id <- strsplit(key, "/")[[1]][[2]]
+    scdata_list[[sample_id]] <- obj
+  }
+
+  return(scdata_list)
+}
+
+# reload_data_from_s3 will reload:
+# * scdata_list for all steps before integration (included)
+# * scdata file for all steps after data integration
+reload_data_from_s3 <- function(pipeline_config, experiment_id, task_name, tasks) {
+  task_names <- names(tasks)
+  integration_index <- match("dataIntegration", task_names)
+  s3 <- paws::s3(config = pipeline_config$aws_config)
+
+  # If the task is after data integration, we need to get scdata from processed_matrix
+  if (match(task_name, task_names) > integration_index) {
+    return(reload_scdata_from_s3(s3))
+  }
+
+  # Otherwise, return scdata_list
+  return(reload_scdata_list_from_s3(s3))
```

Comment:
`reload_scdata_from_s3` and `reload_scdata_list_from_s3` both require `experiment_id` as a parameter, but in these calls I don't see `experiment_id` passed in. How does this work?

Reply:
good catch. I added them